### PR TITLE
Remove requirement that zero length reads need color space or flow si…

### PR DESCRIPTION
**Pull request by nh13**
…gnal data.

See https://github.com/samtools/htsjdk/issues/1174

### CODE REVIEW

File: src/main/java/htsjdk/samtools/SAMRecord.java [hunk 1]
```diff
@@ -2044,35 +2044,7 @@ public int hashCode() {
             ret.addAll(errors);
             if (firstOnly) return ret;
         }
-        // TODO(mccowan): Is this asking "is this the primary alignment"?
```
Comment (nh13) [inline src/main/java/htsjdk/samtools/SAMRecord.java]: @mccowan !

Comment (mccowan) [reply src/main/java/htsjdk/samtools/SAMRecord.java]: Thank you for cleaning up my loose ends!  😁. Hope all is well!

**Comment by yfarjoun** [comment]
I'm not happy about removing this sanity test...

**Comment by nh13** [comment]
@yfarjoun I am not happy about you being not happy.  Lets discuss on the original issue.

**Comment by yfarjoun** [comment]
Well... while I'm not 100% happy about it, I think that if you convert the deletion to a commented code (like all the other skeletons in that method) and add a comment about when it could go back into the code (i.e. when there's a system in place for disabling various errors) I'll relent.... :-)

**Comment by nh13** [comment]
@yfarjoun I don't like leaving dead code, so I guess we both aren't 100% happy, which is a good compromise.  Thank-you!

**Comment by yfarjoun** [comment]
:+1: thanks for compromising.

**Pull request merged**